### PR TITLE
test vaults and reserve indexing

### PR DESCRIPTION
**Pull request by rabi-siddique**
This PR automates the testing of Vault and Reserve entities using GraphQL queries to enhance flexibility in matching expected values. It also removes the test cases previously written with `@subql/testing`, which were restrictive due to issues with dates.

### CODE REVIEW

File: .github/workflows/vaults-and-reserve-metrics-testing.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          echo "Fetching initial logs for all containers..."
+          containers=$(docker ps --format '{{.ID}}')
+          for container in $containers; do
+            echo "Fetching initial logs for container $container..."
+            docker logs $container
+          done
+
+      - name: Set Execute Permission for All Scripts
+        run: chmod +x ./scripts/*.mjs
+
```
Comment (turadg) [inline .github/workflows/vaults-and-reserve-metrics-testing.yml:42]: this should be done in SCM. Git maintains the execution flags. 
```suggestion
```

File: .github/workflows/vaults-and-reserve-metrics-testing.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      - name: Wait for GraphQL server
+        run: sleep 120
+
```
[the remark this answers is not in the document]
Comment (turadg) [reply .github/workflows/vaults-and-reserve-metrics-testing.yml:45]: Helpful to include a comment with what value you've found to be too small

Alternately, `scripts/validateData.mjs` could have a "wait and retry" but this is fine

File: .github/workflows/vaults-and-reserve-metrics-testing.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      - name: Wait for GraphQL server
+        run: sleep 60
+
+      - name: Test VaultManagerGovernance
+        run: ./scripts/validateData.mjs
+        env:
+          entity: vaultManagerGovernances
+          blockHeight: 1212
+          apiUrl: 'http://localhost:3000/'
```
Comment (turadg) [inline .github/workflows/vaults-and-reserve-metrics-testing.yml:51]: please move these repeated env vars to the job level

**Review by frazarshad — APPROVED** [review]

File: .github/workflows/vaults-and-reserve-metrics-testing.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Muneeb147) [reply .github/workflows/vaults-and-reserve-metrics-testing.yml:44]: @rabi-siddique I agree with Turadg's wait and retry suggestion. Such hardcoded sleep times cause flakiness in future. Something like following can be done (as a follow-up)

```
- name: Wait for GraphQL server to be ready
  run: |
    echo "Waiting for GraphQL server to be ready..."
    until curl -s -o /dev/null -w "%{http_code}" http://localhost:4000/graphql | grep -q "200"; do
      echo "GraphQL server is not ready yet. Retrying in 5 seconds..."
      sleep 5
    done
    echo "GraphQL server is ready!"
  shell: bash
```

**Pull request merged**